src/App.tsx

```diff
@@ -72,10 +71,21 @@ function App() {
           JSON.stringify(guessedLetters)
         ) as typeof guessedLetters
         const currentRow = newGuessedLetters[currentRowIndex]
-        // TODO: Check if current row submission already exists in previous row submissions and don't accept if true
+        // DONE: Check if current row submission already exists in previous row submissions and don't accept if true
         // TODO: Check if current row submission is not in the list of words and throw invalid word error if true
+
         const newUserSolution = [...userSolution]
         newUserSolution[currentRowIndex] = currentRow.join('')
+        if (
+          userSolution.some(
+            (solution) => solution === newUserSolution[currentRowIndex]
+          )
+        ) {
+          alert('Word already submitted!')
+          setUserSolution([...userSolution])
```

Comment:
Thanks for the PR 🙏🏻 @moretime-dev 

It looks good. Just one change request:

`setUserSolution([...userSolution])` can be removed. Since you are basically setting the current user solution again

And regarding your question: of if we need to delete the current row, that is not required. Since user might have a typo and it's better if user is allowed to delete the items by himself.